x/ccv/provider/keeper/grpc_query.go

```diff
@@ -232,18 +229,40 @@ func (k Keeper) QueryRegisteredConsumerRewardDenoms(goCtx context.Context, req *
 	}, nil
 }
 
-// TODO (PERMISSIONLESS)
 func (k Keeper) QueryProposedConsumerChainIDs(goCtx context.Context, req *types.QueryProposedChainIDsRequest) (*types.QueryProposedChainIDsResponse, error) {
 	if req == nil {
 		return nil, status.Error(codes.InvalidArgument, "empty request")
 	}
 
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	chains := k.GetAllProposedConsumerChainIDs(ctx)
+	var proposals []govv1.Proposal
+	err := k.govKeeper.Proposals.Walk(ctx, nil, func(proposalId uint64, proposal govv1.Proposal) (stop bool, err error) {
+		proposals = append(proposals, proposal)
```

Comment:
It's safe to execute the logic within the walk without storing the proposals in a separate array.

Reply:
Removed this query; see comment [here](https://github.com/cosmos/interchain-security/pull/2177/files#r1734448221).